app/src/main/java/de/tum/in/tumcampusapp/activities/SurveyActivity.java

```diff
@@ -204,35 +231,35 @@ private void zoomOutanimation(View v) {
         parentView.startAnimation(zoomOut);
         zoomOut.setAnimationListener(new Animation.AnimationListener() {
             @Override
-            public void onAnimationStart(Animation animation) {}
+            public void onAnimationStart(Animation animation) {
+            }
 
             @Override
             public void onAnimationEnd(Animation animation) {
-                parentView.removeAllViews();}
+                parentView.removeAllViews();
+            }
 
             @Override
-            public void onAnimationRepeat(Animation animation) {}});
+            public void onAnimationRepeat(Animation animation) {
+            }
+        });
     }
 
-    //delete button click
-    View.OnClickListener clicks = new View.OnClickListener() {
 
-        @Override
-        public void onClick(final View v) {
-            if (NetUtils.isConnected(getApplicationContext())) {
-                //remove view and delete from database.
-                v.setEnabled(false);
-                int tag = (int) v.getTag();
-                surveyManager.deleteMyOwnQuestion(tag);
-                zoomOutanimation(v);
-                Snackbar.make(findViewById(R.id.drawer_layout), getResources().getString(R.string.question_deleted), Snackbar.LENGTH_LONG).show();
-            } else {
-                restartActivity();
-            }
-        }
 
-    };
-    //get all views by IDs
+    /**
+     * For restarting SurveyAcitivity. Used upon reconnecting to internet after no connection, or when submitting question(s)
+     */
+    private void restartActivity() {
+        Intent i = getIntent();
+        i.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
+        finish();
+        startActivity(i);
```

Comment:
We have just tested the back button after refreshing and it works fine. Is there smth we are not taking into consideration while testing that? Many thanks for your reply in advance.